### PR TITLE
Prefer v1 storage versions

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Switches preferred storage version to v1 for types promoted to v1 <= 1.20.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/cc @deads2k

### CODE REVIEW

File: cmd/kube-apiserver/app/apiextensions.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ func createAPIExtensionsConfig(
 	etcdOptions := *commandOptions.Etcd
 	etcdOptions.StorageConfig.Paging = utilfeature.DefaultFeatureGate.Enabled(features.APIListChunking)
 	etcdOptions.StorageConfig.Codec = apiextensionsapiserver.Codecs.LegacyCodec(v1beta1.SchemeGroupVersion, v1.SchemeGroupVersion)
+	// prefer the more compact serialization (v1beta1) for storage until http://issue.k8s.io/82292 is resolved for objects whose v1 serialization is too big but whose v1beta1 serialization can be stored
```
Comment (enj) [inline cmd/kube-apiserver/app/apiextensions.go:71]: 😬

**Comment by liggitt** [comment]
rebased, retagging

**Pull request merged**